Mask E2E occasionally failed

## Bug Report


### Which version of ShardingSphere did you use?


master - 19758b7c63e394eebd51620484e798ff8f81a1d8

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

### Expected behavior

E2E passed stably.

### Actual behavior

```
Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 60.534 s <<< FAILURE! - in org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT
Error:  assertExecute{AssertionTestParameter}[1]  Time elapsed: 60.513 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: is "first-n=3,last-m=4,replace-char=*"
     but: was "first-n=3,replace-char=*,last-m=4"
	at org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT.assertObjectValue(RQLE2EIT.java:114)
	at org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT.assertRow(RQLE2EIT.java:108)
	at org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT.assertRows(RQLE2EIT.java:98)
	at org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT.assertResultSet(RQLE2EIT.java:74)
	at org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT.assertExecute(RQLE2EIT.java:67)
	at org.apache.shardingsphere.test.e2e.engine.type.RQLE2EIT.assertExecute(RQLE2EIT.java:58)

[INFO] Running org.apache.shardingsphere.test.e2e.engine.type.dml.GeneralDMLE2EIT
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.apache.shardingsphere.test.e2e.engine.type.dml.GeneralDMLE2EIT
[INFO] Running org.apache.shardingsphere.test.e2e.engine.type.dml.BatchDMLE2EIT
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 s - in org.apache.shardingsphere.test.e2e.engine.type.dml.BatchDMLE2EIT
[INFO] Running org.apache.shardingsphere.test.e2e.engine.type.dml.AdditionalDMLE2EIT
Warning:  Tests run: 7, Failures: 0, Errors: 0, Skipped: 7, Time elapsed: 0.001 s - in org.apache.shardingsphere.test.e2e.engine.type.dml.AdditionalDMLE2EIT
[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    RQLE2EIT.assertExecute:58->assertExecute:67->assertResultSet:74->assertRows:98->assertRow:108->assertObjectValue:114 
Expected: is "first-n=3,last-m=4,replace-char=*"
     but: was "first-n=3,replace-char=*,last-m=4"
[INFO] 
Error:  Tests run: 40, Failures: 1, Errors: 0, Skipped: 13
```

### Reason analyze (If you can)

Maybe HashMap was used?

--- a/docs/blog/content/material/2023_02_17_Dynamic_Data_Masking_ShardingSphere_5.3.1_Deep_Dive.en.md
+++ b/docs/blog/content/material/2023_02_17_Dynamic_Data_Masking_ShardingSphere_5.3.1_Deep_Dive.en.md
@@ -250,7 +250,7 @@ mysql> SHOW MASK RULES FROM mask_db;
 +--------+-----------+---------------------------+-----------------------------------+
 | t_user | password  | MD5                       |                                   |
 | t_user | email     | MASK_BEFORE_SPECIAL_CHARS | replace-char=*,special-chars=@    |
-| t_user | telephone | KEEP_FIRST_N_LAST_M       | first-n=3,replace-char=*,last-m=4 |
+| t_user | telephone | KEEP_FIRST_N_LAST_M       | first-n=3,last-m=4,replace-char=* |
 +--------+-----------+---------------------------+-----------------------------------+
 3 rows in set (0.01 sec)
 

--- a/infra/util/src/main/java/org/apache/shardingsphere/infra/util/props/PropertiesConverter.java
+++ b/infra/util/src/main/java/org/apache/shardingsphere/infra/util/props/PropertiesConverter.java
@@ -20,6 +20,7 @@ package org.apache.shardingsphere.infra.util.props;
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
 
+import java.util.LinkedHashMap;
 import java.util.Properties;
 import java.util.stream.Collectors;
 
@@ -36,6 +37,6 @@ public final class PropertiesConverter {
      * @return converted string content
      */
     public static String convert(final Properties props) {
-        return props.entrySet().stream().map(entry -> String.join("=", entry.getKey().toString(), entry.getValue().toString())).collect(Collectors.joining(","));
+        return new LinkedHashMap<>(props).entrySet().stream().map(entry -> String.join("=", entry.getKey().toString(), entry.getValue().toString())).collect(Collectors.joining(","));
     }
 }

--- a/infra/util/src/main/java/org/apache/shardingsphere/infra/util/props/PropertiesConverter.java
+++ b/infra/util/src/main/java/org/apache/shardingsphere/infra/util/props/PropertiesConverter.java
@@ -20,7 +20,6 @@ package org.apache.shardingsphere.infra.util.props;
 import lombok.AccessLevel;
 import lombok.NoArgsConstructor;
 
-import java.util.LinkedHashMap;
 import java.util.Properties;
 import java.util.stream.Collectors;
 
@@ -37,6 +36,6 @@ public final class PropertiesConverter {
      * @return converted string content
      */
     public static String convert(final Properties props) {
-        return new LinkedHashMap<>(props).entrySet().stream().map(entry -> String.join("=", entry.getKey().toString(), entry.getValue().toString())).collect(Collectors.joining(","));
+        return props.entrySet().stream().map(entry -> String.join("=", entry.getKey().toString(), entry.getValue().toString())).sorted().collect(Collectors.joining(","));
     }
 }

--- a/test/e2e/sql/src/test/resources/cases/rdl/dataset/rdl_empty_rules/alter_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rdl/dataset/rdl_empty_rules/alter_mask_rule.xml
@@ -22,5 +22,5 @@
         <column name="algorithm_type" />
         <column name="algorithm_props" />
     </metadata>
-    <row values="t_mask| order_id| MASK_FROM_X_TO_Y| to-y=2,from-x=1,replace-char=*" />
+    <row values="t_mask| order_id| MASK_FROM_X_TO_Y| from-x=1,replace-char=*,to-y=2" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask/cluster/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask/cluster/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
     <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask/standalone/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask/standalone/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
-    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,replace-char=*,last-m=4" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
+    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt/cluster/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt/cluster/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
     <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
-    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,replace-char=*,last-m=4" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
+    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt/standalone/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt/standalone/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
-    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,replace-char=*,last-m=4" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
+    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt_sharding/cluster/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt_sharding/cluster/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
     <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt_sharding/standalone/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_encrypt_sharding/standalone/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
-    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,replace-char=*,last-m=4" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
+    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_sharding/cluster/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_sharding/cluster/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
     <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

--- a/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_sharding/standalone/show_mask_rule.xml
+++ b/test/e2e/sql/src/test/resources/cases/rql/dataset/mask_sharding/standalone/show_mask_rule.xml
@@ -23,6 +23,6 @@
         <column name="algorithm_props" />
     </metadata>
     <row values="t_user| password| MD5| " />
-    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| special-chars=@,replace-char=*" />
-    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,replace-char=*,last-m=4" />
+    <row values="t_user| email| MASK_BEFORE_SPECIAL_CHARS| replace-char=*,special-chars=@" />
+    <row values="t_user| telephone| KEEP_FIRST_N_LAST_M| first-n=3,last-m=4,replace-char=*" />
 </dataset>

